seal.go

```diff
@@ -0,0 +1,347 @@
+	}
+
+	// Use the PCR digests and NV index names to generate a single signed dynamic authorization policy digest
+	policyParams := dynamicPolicyComputeParams{
+		key:                  authKey,
+		signAlg:              signAlg,
+		pcrValues:            pcrValues,
+		policyCountIndexName: countIndexName,
+		policyCount:          nextPolicyCount}
+
+	policyData, err := computeDynamicPolicy(alg, &policyParams)
+	if err != nil {
+		return nil, xerrors.Errorf("cannot compute dynamic authorization policy: %w", err)
+	}
+
+	return policyData, nil
+}
+
+// KeyCreationParams provides the arguments for SealKeyToTPM.
+type KeyCreationParams struct {
+	PinHandle tpm2.Handle // Handle at which to create a NV index for PIN support
+}
+
+// SealKeyToTPM seals the supplied disk encryption key to the storage hierarchy of the TPM. The sealed key object and associated
+// metadata that is required during early boot in order to unseal the key again and unlock the associated encrypted volume is written
+// to a file at the path specified by keyPath. Additional data that is required in order to update the authorization policy for the
+// sealed key is written to a file at the path specified by policyUpdatePath. This file must live inside the encrypted volume
+// protected by the sealed key.
+//
+// The supplied key must be 32 bytes long. An error will be returned if it isn't.
+//
+// This function requires knowledge of the authorization value for the storage hierarchy, which must be provided by calling
+// TPMConnection.OwnerHandleContext().SetAuthValue() prior to calling this function. If the provided authorization value is incorrect,
+// a AuthFailError error will be returned.
+//
+// If the TPM is not correctly provisioned, a ErrTPMProvisioning error will be returned. In this case, ProvisionTPM must be called
+// before proceeding.
+//
+// This function expects there to be no files at the specified paths. If either path references a file that already exists, a wrapped
+// *os.PathError error will be returned with an underlying error of syscall.EEXIST. A wrapped *os.PathError error will be returned if
+// either file cannot be created and opened for writing.
+//
+// The caller is expected to provide a handle at which a NV index should be created via the PinHandle filed of the KeyCreationParams
+// struct. If the handle is already in use, a TPMResourceExistsError error will be returned. The handle must be a valid NV index
+// handle (MSO == 0x01), and the choice of handle should take in to consideration the reserved indices from the "Registry of reserved
+// TPM 2.0 handles and localities" specification. It is recommended that the handle is in the block reserved for owner objects
+// (0x01800000 - 0x01bfffff).
+//
+// The key will be protected with a PCR policy computed from the supplied PCRProtectionProfile.
+func SealKeyToTPM(tpm *TPMConnection, keyPath, policyUpdatePath string, params *KeyCreationParams, pcrProfile *PCRProtectionProfile, key []byte) error {
```

Comment:
I would expect key []byte to be the second argument, it's conceptually the main input